### ISSUE TITLE
[Edit text] Can't click edit after send message

### ISSUE THREAD

**Issue by hieuhoang998** [linagora/Twake-Mobile#255]
#Precondition:
version:2.2.2 twake app
device: android: oppo reno 4 
#Step
1.Open twake app
2.Log in successfully
3.Send a message
4.Click message then click edit

#Actual result: Edit function not working, when clicking edit tab, menu tab don't disappear
#Expect result: Edit function working properly

![processed (6)](https://user-images.githubusercontent.com/54783033/106715675-fbdd1c00-662f-11eb-9537-92a1e2b2d193.jpeg)

**Comment by bobs4462** [comment]
The issue is fixed, but only partially. You need to wait a bit (usually less than a second), before the message has been fully synced with remote server, and can edit message only after that.

**Comment by tlle14** [comment]
Thank @bobs4462 
I can easily select `edit` button after sending messages on Samsung A51.
Could you please confirm on your device? @hieuhoang998

**Comment by tlle14** [comment]
Hi @bobs4462 
it still is an issue with the new messages, but quite randomly.
I should decrease its priority

**Comment by bobs4462** [comment]
Ok, I'll try to poke around and catch this bug, if you will find a way to reproduce it, let me know, please.

**Comment by bobs4462** [comment]
I think that you should wait a half a second longer. The problem you are encountering is the following:

- When you send a message, the app creates a dummy message locally, without ID (we cannot assign ID ourselves, because it should be generated by the backend). 
- Then a second or two later the server responds with the id of newly created message, and the application updates this newly created message with the actual ID. 
- Without this actual ID there's no way to edit or delete the message, because when you do, you have to tell the server: **hey here the message with the ID you gave me, I want to edit/delete it**. 
- As you can see, if you have no ID you won't be able to tell the server which message you want to operate on.

Hope that makes sense :).

I know that this can be an annoying issue, but it is sort of an architectural issue and cannot be easily fixed right now.

**Comment by tlle14** [comment]
Thank @bobs4462 for a detailed explanation. 
We should fix this issue later because it has a low priority

**Comment by tlle14** [comment]
Hi @bobs4462 how do you think about this solution? 
While the message is still the text in local, it should be displayed with opacity. Users may not open the context menu with this kind of message.
After it has an ID, it will display clearly. So users can manipulate this message.

**Comment by bobs4462** [comment]
Perfect solution, actually it's one the obvious solutions that I had in mind for quite some time. 
@BastiaanVanGaalen Can you please think it over from UX perspective?

**Comment by BastiaanVanGaalen** [comment]
@bobs4462 @tlle14 I would show the "deferred delivery icon" next to the bubble so user are clear the message is being sent/being processed. Moreover we have to enable users edit messages even if they haven't still synced with ie allow local adjustments and keep them in core device memory till and do all such manipulation on the background so users don't even waste their time figuring out "delivery time lags". Is it doable at all?

**Comment by bobs4462** [comment]
Yes, it's doable, and IMHO that's the way a mobile messenger app should behave. But...
But there's always buts :)

The problem is that currently state management of the application evolved into an Italian spaghetti pasta, and requires thorough reevaluation and redesign. Without that solid foundation (top-notch state management) it's going to take forever to implement any somewhat complex feature.

**Comment by tlle14** [comment]
Retest pass! 
When the message is not successful delivered, the context menu will not display.
